ObservatoryHerald/VoiceAzureCacheManager.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Xml;
+using Microsoft.CognitiveServices.Speech;
+using System.Collections.ObjectModel;
+
+namespace Observatory.Herald
+{
+    internal class VoiceAzureCacheManager
+    {
+        private string azureKey;
+        private DirectoryInfo cacheLocation;
+        private Dictionary<string, string> cacheIndex;
+        private SpeechConfig speechConfig;
+        private SpeechSynthesizer speech;
+
+
+        private string cacheIndexFile
+        {
+            get => cacheLocation.FullName + "\\VoiceIndex.json";
+        }
+
+        internal VoiceAzureCacheManager(HeraldSettings settings, HttpClient httpClient)
+        {
+            cacheLocation = new(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "\\ObservatoryCore\\ObservatoryHerald\\");
+
+            if (!Directory.Exists(cacheLocation.FullName))
+            {
+                Directory.CreateDirectory(cacheLocation.FullName);
+            }
+
+            if (File.Exists(cacheIndexFile))
+            {
+                try
+                {
+                    cacheIndex = JsonSerializer.Deserialize<Dictionary<string, string>>(cacheIndexFile);
+                }
+                catch
+                {
+                    cacheIndex = new();
+                }
+            }
+            else
+            {
+                cacheIndex = new();
+            }
+
+            azureKey = GetAzureKey(settings, httpClient);
+            speechConfig = SpeechConfig.FromSubscription(azureKey, "eastus");
```

Comment:
Should "eastus" also be configurable?

Reply:
According to MS's documentation the region used here is expected to match the region of the Azure account, not the user. It probably should be configurable in the event that someone chooses to override with their own key though, so I'll keep that in mind.